src/schema/facet.rs

```diff
@@ -75,11 +76,47 @@ impl Facet {
     /// It is conceptually, if one of the steps of this path
     /// contains a `/` or a `\`, it should be escaped
     /// using an anti-slash `/`.
-    pub fn from_text<T>(path: &T) -> Facet
+    pub fn from_text<T>(path: &T) -> Result<Facet, TantivyError>
     where
         T: ?Sized + AsRef<str>,
     {
-        From::from(path)
+        #[derive(Copy, Clone)]
+        enum State {
+            Escaped,
+            Idle,
+        }
+        let path_ = path.as_ref();
```

Comment:
`path_ref` or `path` [although @fulmicoton doesn't like shadowing :)] would be a better name

Reply:
path_ref please.

Reply:
I don't have any preference here.